tests/test_minimal/test_tools/test_expand_paths.py

```diff
@@ -421,3 +425,96 @@ def test_expand_paths_ibl(tmpdir):
 
     tc = unittest.TestCase()
     tc.assertCountEqual(path_expansion_results, expected)
+
+
+# contruct_path_template tests
+
+
+test_cases = [
+    dict(
+        test_input=dict(
+            path="/data/subject456/session123/file.txt", metadata=dict(subject_id="subject456", session_id="session123")
+        ),
+        expected="/data/{subject_id}/{session_id}/file.txt",
+    ),
+    dict(
+        test_input=dict(
+            path="/data/subject789/session456/image.txt",
+            metadata=dict(subject_id="subject789", session_id="session456", file_type="image"),
+        ),
+        expected="/data/{subject_id}/{session_id}/{file_type}.txt",
+    ),
+    dict(
+        test_input=dict(
+            path="/SenzaiY/YMV01/YMV01_170818/YMV01_170818.eeg", metadata=dict(session_id="170818", subject_id="YMV01")
+        ),
+        expected="/SenzaiY/{subject_id}/{subject_id}_{session_id}/{subject_id}_{session_id}.eeg",
+    ),
+    dict(
+        test_input=dict(path="/SenzaiY/YMV01/YMV01_170818/", metadata=dict(session_id="170818", subject_id="YMV01")),
+        expected="/SenzaiY/{subject_id}/{subject_id}_{session_id}/",
+    ),
+    dict(
+        test_input=dict(
+            path="/steinmetzlab/Subjects/NR_0017/2022-03-22/001/raw_video_data/_iblrig_leftCamera.raw.6252a2f0-c10f-4e49-b085-75749ba29c35.mp4",
+            metadata=dict(
+                session_id="001",
+                subject_id="NR_0017",
+                camera_id="leftCamera.raw.6252a2f0-c10f-4e49-b085-75749ba29c35",
+                session_date="2022-03-22",
+            ),
+        ),
+        expected="/steinmetzlab/Subjects/{subject_id}/{session_date}/{session_id}/raw_video_data/_iblrig_{camera_id}.mp4",
+    ),
```

Comment:
```suggestion
    dict(
        test_input=dict(
            path="/steinmetzlab/Subjects/NR_0017/2022-03-22/001/raw_video_data/_iblrig_leftCamera.raw.6252a2f0-c10f-4e49-b085-75749ba29c35.mp4",
            metadata=dict(
                session_id="001",
                subject_id="NR_0017",
                camera_id="leftCamera.raw.6252a2f0-c10f-4e49-b085-75749ba29c35",
                session_date="2022-03-22",
            ),
        ),
        expected="/steinmetzlab/Subjects/{subject_id}/{session_date}/{session_id}/raw_video_data/_iblrig_{camera_id}.mp4",
    ),
    dict(
        test_input=dict(
            path="/steinmetzlab/Subjects/NR_0017/2022-03-22/001/raw_video_data/_iblrig_leftCamera.raw.6252a2f0-c10f-4e49-b085-75749ba29c35.mp4",
            metadata=dict(
                subject_id="NR_0017",
                session_date="2022-03-22",
                camera_number="001",
                video_id="6252a2f0-c10f-4e49-b085-75749ba29c35",
            ),
        ),
        expected="/steinmetzlab/Subjects/{subject_id}/{session_date}/{camera_number}/raw_video_data/_iblrig_{video_id}.mp4",
    ),
```

At least one more little test case here; the intention being that programmatically there exists a map from from the `video_id` hash to the actual `session_id` (just the way they did things in the IBL dataset), though the actual `eid` (as they refer to it) is not a part of the video filename